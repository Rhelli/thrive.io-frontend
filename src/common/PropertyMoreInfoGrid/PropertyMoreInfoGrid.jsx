import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faMoneyBillWaveAlt, faWifi, faChair, faLeaf, faCat, faSmoking, faWheelchair,
  faParking,
} from '@fortawesome/free-solid-svg-icons';
import { arrayToString } from '../../utils/managePropertiesUtils';
import styles from './PropertyMoreInfoGrid.module.scss';

const PropertyMoreInfoGrid = ({ property }) => {
  const {
    bills, disabledAccess, furnished, internet, outsideArea, parking, pets, smoking,
  } = property;

  return (
    <div className={styles.additionalInfoContainer}>
      <h2>Additional Information</h2>
      <div className={styles.infoSegmentsContainer}>
        <div>
          <span>
            <FontAwesomeIcon icon={faMoneyBillWaveAlt} />
            <h4>Bills?</h4>
            {
              bills === 'Included' ? (
                <p className={styles.greenText}>{bills}</p>
              ) : (
                <p className={styles.redText}>{bills}</p>
              )
            }
          </span>
          <span>
            <FontAwesomeIcon icon={faWifi} />
            <h4>Internet?</h4>
            {
              internet === 'Internet Included' ? (
                <p className={styles.greenText}>{internet}</p>
              ) : (
                <p className={styles.redText}>{internet}</p>
              )
            }
          </span>
        </div>
        <div>
          <span>
            <FontAwesomeIcon icon={faChair} />
            <h4>Furnished?</h4>
            {
              furnished === 'Furnished' ? (
                <p className={styles.greenText}>{furnished}</p>
              ) : (
                <p className={styles.redText}>{furnished}</p>
              )
            }
          </span>
          <span>
            <FontAwesomeIcon icon={faLeaf} />
            <h4>Outside Areas?</h4>
            {
              !outsideArea.length ? (
                <p className={styles.redText}>None</p>
              ) : (
                <p className={styles.greenText}>{arrayToString(outsideArea)}</p>
              )
            }
          </span>
        </div>
        <div>
          <span>
            <FontAwesomeIcon icon={faCat} />
            <h4>Pets?</h4>
            {
              pets.length === 1 && pets[0] === 'None' ? (
                <p className={styles.neutralText}>None</p>
              ) : (
                <p className={styles.greenText}>{arrayToString(pets)}</p>
              )
            }
          </span>
          <span>
            <FontAwesomeIcon icon={faSmoking} />
            <h4>Smoking?</h4>
            {
              smoking === 'Smoking' ? (
                <p className={styles.redText}>{smoking}</p>
              ) : smoking === 'Any' ? (
                <p className={styles.neutralText}>{smoking}</p>
              ) : (
                <p className={styles.greenText}>{smoking}</p>
              )
            }
          </span>
        </div>
        <div>
          <span>
            <FontAwesomeIcon icon={faWheelchair} />
            <h4>Disabled Access?</h4>
            {
              disabledAccess === 'Disabled Access' ? (
                <p className={styles.greenText}>{disabledAccess}</p>
              ) : (
                <p className={styles.neutralText}>{disabledAccess}</p>
              )
            }
          </span>
          <span>
            <FontAwesomeIcon icon={faParking} />
            <h4>Parking?</h4>
            {
              parking === 'No Parking' ? (
                <p className={styles.redText}>{parking}</p>
              ) : (
                <p className={styles.redText}>{parking}</p>
              )
            }
          </span>
        </div>
      </div>
    </div>
  );
};

PropertyMoreInfoGrid.propTypes = {
  property: PropTypes.shape({
    bills: PropTypes.string,
    disabledAccess: PropTypes.string,
    furnished: PropTypes.string,
    outsideArea: PropTypes.arrayOf(PropTypes.string),
    parking: PropTypes.string,
    pets: PropTypes.arrayOf(PropTypes.string),
    smoking: PropTypes.string,
    internet: PropTypes.string,
    length: PropTypes.number,
  }).isRequired,
};

export default PropertyMoreInfoGrid;
