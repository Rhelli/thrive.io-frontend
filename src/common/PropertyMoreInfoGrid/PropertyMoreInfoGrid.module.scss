.additionalInfoContainer {
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  align-items: center;
  width: 100%;
  padding: 10px 20px;
  
  h2 {
    width: 100%;
    text-align: left;
    margin-bottom: 20px;
  }

  .infoSegmentsContainer {
    display: flex;
    flex-flow: column nowrap;
    align-items: center;
    justify-content: center;
    width: 100%;

    div {
      display: flex;
      flex-flow: row nowrap;
      justify-content: center;
      width: 100%;
      border: 1px solid blue;

      span {
        display: flex;
        flex-flow: column nowrap;
        align-items: center;
        justify-content: center;
        width: 50%;
        height: 100px;
        border: 1px solid red;

        svg {
          font-size: 32px;
        }

        h4 {
          margin: 10px 0 2px 0;
          font-size: 16px;
        }

        p {
          font-size: 14px;
        }

        .greenText,
        .redText,
        .neutralText {
          padding: 2px 4px;
          border-radius: 3px;
        }

        .greenText {
          background-color: green;
        }

        .redText {
          background-color: darkred;
        }

        .neutralText {
          background-color: #707070;
        }
      }

      .disabledSpan {
        width: 100%;
      }
    }
  }
}