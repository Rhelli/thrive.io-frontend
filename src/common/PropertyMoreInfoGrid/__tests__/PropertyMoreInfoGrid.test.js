import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import PropertyMoreInfoGrid from '../PropertyMoreInfoGrid';

Enzyme.configure({ adapter: new Adapter() });

const property = {
  id: 2,
  title: 'Non in beatae unde nostrum hic.',
  blurb: 'Et unde dignissimos qui eveniet ipsa blanditiis quasi sunt dicta alias est ut eaque iusto excepturi ea voluptatem tempora officiis voluptatem sed amet.',
  address: '999 Graham Crossing',
  town: 'Bayswater',
  postcode: 'SW6 1PW',
  price: 1475,
  deposit: 1042,
  bills: 'Not Included',
  furnished: 'Furnished',
  parking: 'Parking',
  outsideArea: [],
  disabledAccess: 'No Disabled Access',
  internet: 'No Internet Included',
  occupantCount: 4,
  roomCount: 5,
  minAuge: 29,
  maxAuge: 59,
  smoking: 'Any',
  pets: ['Rodents'],
  genders: ['Male', 'Female', 'Male', 'Transgender'],
  occupations: ['Professional', 'Professional', 'Professional', 'Professional'],
};

const propertyMoreInfoGrid = (
  <PropertyMoreInfoGrid property={property} />
);

describe('The appearance and functionality of the propertyListGrid component', () => {
  const container = mount(propertyMoreInfoGrid);
  const grid = container.find('.infoSegmentsContainer');

  it('Matches the previous snapshot', () => {
    const tree = renderer.create(propertyMoreInfoGrid).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Has a master container with the correct attributes and a title', () => {
    expect(container.childAt(0).hasClass('additionalInfoContainer')).toBe(true);
    expect(container.childAt(0).childAt(0).text()).toBe('Additional Information');
  });

  it('Displays the bills correctly', () => {
    expect(grid.childAt(0).childAt(0).childAt(1).text()).toBe('Bills?');
    expect(grid.childAt(0).childAt(0).childAt(2).text()).toBe('Not Included');
  });

  it('Displays the internet info correctly', () => {
    expect(grid.childAt(0).childAt(1).childAt(1).text()).toBe('Internet?');
    expect(grid.childAt(0).childAt(1).childAt(2).text()).toBe('No Internet Included');
  });

  it('Displays the furnished info correctly', () => {
    expect(grid.childAt(1).childAt(0).childAt(1).text('Furnished?'));
    expect(grid.childAt(1).childAt(0).childAt(2).text('Furnished'));
  });

  it('Displays the outside area info correctly', () => {
    expect(grid.childAt(1).childAt(1).childAt(1).text()).toBe('Outside Areas?');
    expect(grid.childAt(1).childAt(1).childAt(2).text()).toBe('None');
  });

  it('Displays the pets info correctly', () => {
    expect(grid.childAt(2).childAt(0).childAt(1).text()).toBe('Pets?');
    expect(grid.childAt(2).childAt(0).childAt(2).text()).toBe('Rodents');
  });

  it('Displays the smoking info correctly', () => {
    expect(grid.childAt(2).childAt(1).childAt(1).text()).toBe('Smoking?');
    expect(grid.childAt(2).childAt(1).childAt(2).text()).toBe('Any');
  });

  it('Displays the disabled access info correctly', () => {
    expect(grid.childAt(3).childAt(0).childAt(1).text()).toBe('Disabled Access?');
    expect(grid.childAt(3).childAt(0).childAt(2).text()).toBe('No Disabled Access');
  });

  it('Displays the parking info correctly', () => {
    expect(grid.childAt(3).childAt(1).childAt(1).text()).toBe('Parking?');
    expect(grid.childAt(3).childAt(1).childAt(2).text()).toBe('Parking');
  });
});
