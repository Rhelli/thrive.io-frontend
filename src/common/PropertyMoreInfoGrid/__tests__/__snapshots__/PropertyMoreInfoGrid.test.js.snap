// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The appearance and functionality of the propertyListGrid component Matches the previous snapshot 1`] = `
<div
  className="additionalInfoContainer"
>
  <h2>
    Additional Information
  </h2>
  <div
    className="infoSegmentsContainer"
  >
    <div>
      <span>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-money-bill-wave-alt fa-w-20 "
          data-icon="money-bill-wave-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M621.16 54.46C582.37 38.19 543.55 32 504.75 32c-123.17-.01-246.33 62.34-369.5 62.34-30.89 0-61.76-3.92-92.65-13.72-3.47-1.1-6.95-1.62-10.35-1.62C15.04 79 0 92.32 0 110.81v317.26c0 12.63 7.23 24.6 18.84 29.46C57.63 473.81 96.45 480 135.25 480c123.17 0 246.34-62.35 369.51-62.35 30.89 0 61.76 3.92 92.65 13.72 3.47 1.1 6.95 1.62 10.35 1.62 17.21 0 32.25-13.32 32.25-31.81V83.93c-.01-12.64-7.24-24.6-18.85-29.47zM320 352c-44.19 0-80-42.99-80-96 0-53.02 35.82-96 80-96s80 42.98 80 96c0 53.03-35.83 96-80 96z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <h4>
          Bills?
        </h4>
        <p
          className="redText"
        >
          Not Included
        </p>
      </span>
      <span>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-wifi fa-w-20 "
          data-icon="wifi"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M634.91 154.88C457.74-8.99 182.19-8.93 5.09 154.88c-6.66 6.16-6.79 16.59-.35 22.98l34.24 33.97c6.14 6.1 16.02 6.23 22.4.38 145.92-133.68 371.3-133.71 517.25 0 6.38 5.85 16.26 5.71 22.4-.38l34.24-33.97c6.43-6.39 6.3-16.82-.36-22.98zM320 352c-35.35 0-64 28.65-64 64s28.65 64 64 64 64-28.65 64-64-28.65-64-64-64zm202.67-83.59c-115.26-101.93-290.21-101.82-405.34 0-6.9 6.1-7.12 16.69-.57 23.15l34.44 33.99c6 5.92 15.66 6.32 22.05.8 83.95-72.57 209.74-72.41 293.49 0 6.39 5.52 16.05 5.13 22.05-.8l34.44-33.99c6.56-6.46 6.33-17.06-.56-23.15z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <h4>
          Internet?
        </h4>
        <p
          className="redText"
        >
          No Internet Included
        </p>
      </span>
    </div>
    <div>
      <span>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-chair fa-w-14 "
          data-icon="chair"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M112 128c0-29.5 16.2-55 40-68.9V256h48V48h48v208h48V59.1c23.8 13.9 40 39.4 40 68.9v128h48V128C384 57.3 326.7 0 256 0h-64C121.3 0 64 57.3 64 128v128h48zm334.3 213.9l-10.7-32c-4.4-13.1-16.6-21.9-30.4-21.9H42.7c-13.8 0-26 8.8-30.4 21.9l-10.7 32C-5.2 362.6 10.2 384 32 384v112c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16V384h256v112c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16V384c21.8 0 37.2-21.4 30.3-42.1z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <h4>
          Furnished?
        </h4>
        <p
          className="greenText"
        >
          Furnished
        </p>
      </span>
      <span>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-leaf fa-w-18 "
          data-icon="leaf"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 576 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M546.2 9.7c-5.6-12.5-21.6-13-28.3-1.2C486.9 62.4 431.4 96 368 96h-80C182 96 96 182 96 288c0 7 .8 13.7 1.5 20.5C161.3 262.8 253.4 224 384 224c8.8 0 16 7.2 16 16s-7.2 16-16 16C132.6 256 26 410.1 2.4 468c-6.6 16.3 1.2 34.9 17.5 41.6 16.4 6.8 35-1.1 41.8-17.3 1.5-3.6 20.9-47.9 71.9-90.6 32.4 43.9 94 85.8 174.9 77.2C465.5 467.5 576 326.7 576 154.3c0-50.2-10.8-102.2-29.8-144.6z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <h4>
          Outside Areas?
        </h4>
        <p
          className="redText"
        >
          None
        </p>
      </span>
    </div>
    <div>
      <span>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-cat fa-w-16 "
          data-icon="cat"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M290.59 192c-20.18 0-106.82 1.98-162.59 85.95V192c0-52.94-43.06-96-96-96-17.67 0-32 14.33-32 32s14.33 32 32 32c17.64 0 32 14.36 32 32v256c0 35.3 28.7 64 64 64h176c8.84 0 16-7.16 16-16v-16c0-17.67-14.33-32-32-32h-32l128-96v144c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V289.86c-10.29 2.67-20.89 4.54-32 4.54-61.81 0-113.52-44.05-125.41-102.4zM448 96h-64l-64-64v134.4c0 53.02 42.98 96 96 96s96-42.98 96-96V32l-64 64zm-72 80c-8.84 0-16-7.16-16-16s7.16-16 16-16 16 7.16 16 16-7.16 16-16 16zm80 0c-8.84 0-16-7.16-16-16s7.16-16 16-16 16 7.16 16 16-7.16 16-16 16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <h4>
          Pets?
        </h4>
        <p
          className="greenText"
        >
          Rodents
        </p>
      </span>
      <span>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-smoking fa-w-20 "
          data-icon="smoking"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 640 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M632 352h-48c-4.4 0-8 3.6-8 8v144c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V360c0-4.4-3.6-8-8-8zM553.3 87.1c-5.7-3.8-9.3-10-9.3-16.8V8c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v62.3c0 22 10.2 43.4 28.6 55.4 42.2 27.3 67.4 73.8 67.4 124V280c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8v-30.3c0-65.5-32.4-126.2-86.7-162.6zM432 352H48c-26.5 0-48 21.5-48 48v64c0 26.5 21.5 48 48 48h384c8.8 0 16-7.2 16-16V368c0-8.8-7.2-16-16-16zm-32 112H224v-64h176v64zm87.7-322.4C463.8 125 448 99.3 448 70.3V8c0-4.4-3.6-8-8-8h-48c-4.4 0-8 3.6-8 8v66.4c0 43.7 24.6 81.6 60.3 106.7 22.4 15.7 35.7 41.2 35.7 68.6V280c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8v-30.3c0-43.3-21-83.4-56.3-108.1zM536 352h-48c-4.4 0-8 3.6-8 8v144c0 4.4 3.6 8 8 8h48c4.4 0 8-3.6 8-8V360c0-4.4-3.6-8-8-8z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <h4>
          Smoking?
        </h4>
        <p
          className="neutralText"
        >
          Any
        </p>
      </span>
    </div>
    <div>
      <span>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-wheelchair fa-w-16 "
          data-icon="wheelchair"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M496.101 385.669l14.227 28.663c3.929 7.915.697 17.516-7.218 21.445l-65.465 32.886c-16.049 7.967-35.556 1.194-43.189-15.055L331.679 320H192c-15.925 0-29.426-11.71-31.679-27.475C126.433 55.308 128.38 70.044 128 64c0-36.358 30.318-65.635 67.052-63.929 33.271 1.545 60.048 28.905 60.925 62.201.868 32.933-23.152 60.423-54.608 65.039l4.67 32.69H336c8.837 0 16 7.163 16 16v32c0 8.837-7.163 16-16 16H215.182l4.572 32H352a32 32 0 0 1 28.962 18.392L438.477 396.8l36.178-18.349c7.915-3.929 17.517-.697 21.446 7.218zM311.358 352h-24.506c-7.788 54.204-54.528 96-110.852 96-61.757 0-112-50.243-112-112 0-41.505 22.694-77.809 56.324-97.156-3.712-25.965-6.844-47.86-9.488-66.333C45.956 198.464 0 261.963 0 336c0 97.047 78.953 176 176 176 71.87 0 133.806-43.308 161.11-105.192L311.358 352z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <h4>
          Disabled Access?
        </h4>
        <p
          className="neutralText"
        >
          No Disabled Access
        </p>
      </span>
      <span>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-parking fa-w-14 "
          data-icon="parking"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M400 32H48C21.5 32 0 53.5 0 80v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V80c0-26.5-21.5-48-48-48zM240 320h-48v48c0 8.8-7.2 16-16 16h-32c-8.8 0-16-7.2-16-16V144c0-8.8 7.2-16 16-16h96c52.9 0 96 43.1 96 96s-43.1 96-96 96zm0-128h-48v64h48c17.6 0 32-14.4 32-32s-14.4-32-32-32z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        <h4>
          Parking?
        </h4>
        <p
          className="redText"
        >
          Parking
        </p>
      </span>
    </div>
  </div>
</div>
`;
